### DISCUSSION TITLE
Insert transaction via API do not trigger "updated_at" rules

### DISCUSSION THREAD

**Discussion by Timmy93** [firefly-iii/firefly-iii#8284] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I'm trying to import transaction from Splitwise to Firefly, but when I use the `create transaction` api, no rules are triggered.
The field `apply_rules` is set to `True` and I tought this should be enough.

Am I missing the meaning of the `apply_rule` field? Should I manually retrigger all rules via api?

This is a (parametrized) example of the sent json:

```python
data = {
  "error_if_duplicate_hash": True,
  "apply_rules": True,
  "transactions": [
    {
      "type": "withdrawal",
      "date": date,
      "amount": amount,
      "description": description,
      "category_name": category,
      "source_name": source,
      "destination_name": dest,
      "tags": tag,
      "notes": f"Import from Splitwise\n\n{message}",
      "external_id": sw_id
    }
  ]
}
```

Thank you for your support

### Debug information

Debug information generated at 2023-12-22 18:51:20 for Firefly III version **v6.1.0**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>v6.1.0 / v2.0.12 / 22 (exp. 22)</td></tr><tr><td>PHP version</td><td>8.3.0 / apache2handler / Linux</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, pgsql, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>852, base <span>#</span>67</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>Europe/Rome + Europe/Rome</td></tr><tr><td>App environment</td><td>local, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>notice, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>it_IT + equal</td></tr><tr><td>Trusted proxies</td><td>**</td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>2023-12-22 00:00:00 (18 hours ago)</td></tr><tr><td>Mailer</td><td>log</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 2</td></tr><tr><td>User flags</td><td><span title="Has rules">:wrench:</span> <span title="Has recurring transactions">:clock130:</span> <span title="Has subscriptions">:email:</span></td></tr><tr><td>Session start</td><td>2023-01-01 00:00:00</td></tr><tr><td>Session end</td><td>2023-12-22 23:59:59</td></tr><tr><td>View range</td><td>YTD</td></tr><tr><td>User language</td><td>en_US</td></tr><tr><td>User locale</td><td>en_US</td></tr><tr><td>Locale(s) supported</td><td>en_US.utf8: :white_check_mark:<br>en_US.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:121.0) Gecko/20100101 Firefox/121.0</td></tr></tbody></table>

### Expected behaviour

When I insert a new transaction with the flag `apply_rule` to True, the rule engine process the new transaction.

### Steps to reproduce

1. Create a dummy rule that matches the Description content ("Description contains...")
2. Insert a new transaction **via API**

**Comment by JC5** [comment]
That should be enough, this is how Firefly III does it as well. Keep in mind, valid JSON is lowercase `true`, but I'm sure it's parsed as such.

**Comment by JC5** [comment]
The Firefly III logs should tell you if a rule is triggered, please consult those as well.

https://docs.firefly-iii.org/how-to/general/debug/

**Reply by alexschlueter** [reply]
Hi, I'm having the same problem with rules triggered by other rules not getting executed. Is that not supported? Will it ever be? Thank you :-)

**Reply by JC5** [reply]
No, it will not be supported.

**Comment by Timmy93** [comment]
Thank you I just turned on the logs and I figured out whats the problem:
there are 2 rules triggering:
1. The first triggers with a new transaction and set the Category based on Description (I confirm it is working correctly)
2. The second triggers on update of a transaction and set the Budget based on Category (This rule is not working)

So now the real question is: is it possible to chain rule triggers?

**Reply by JC5** [reply]
Sorry, I missed this entirely. But no. Rules do not chain.
